### PR TITLE
tackling failing test cases - dev deployment

### PR DESCRIPTION
**Pull request by parvinba**

### CODE REVIEW

**Review by farrukhishaq — APPROVED** [review]

File: client/pages/sections/protocols/steps.js [hunk 1]
```diff
@@ -247,7 +247,7 @@ class Step extends Component {
       >
         <NewComments comments={relevantComments} />
         {
-          !values.deleted && <StepBadge fields={values} changeFieldPrefix={changeFieldPrefix} protocolId={protocol.id} position={index}/>
+          !values.deleted && !pdf && <StepBadge fields={values} changeFieldPrefix={changeFieldPrefix} protocolId={protocol.id} position={index}/>
```
Comment (jeff-horton-ho-sas) [inline client/pages/sections/protocols/steps.js:250]: The ChangeBadge code handles this by checking if the arrays are valid https://github.com/UKHomeOffice/asl-projects/blob/master/client/components/changed-badge.js#L23-L32. I don't think we can rely on all of them being set if we're not rendering a pdf - so I think it would be better to check in the StepBadge and return null if the previous step arrays aren't defined.

**Review by jeff-horton-ho-sas — CHANGES_REQUESTED** [review]
Whilst the tests are only failing on the PDF I don't trust that to be the only edge case where this won't work. See comment.

File: client/components/step-badge.js [hunk 2]
```diff
@@ -7,6 +7,9 @@ const changeFields = (step, prefix) => step.reusable ? [ `reusableSteps.${step.r
 
 export default function StepBadge(props) {
   const { previous, steps, firstSteps, grantedSteps } = useSelector(state => state.application.previousProtocols);
+  if (!previous || !steps || !firstSteps || !grantedSteps) {
+    return <></>;
```
Comment (jeff-horton-ho-sas) [inline client/components/step-badge.js:11]:
```suggestion
    return null;
```

**Review by jeff-horton-ho-sas — APPROVED** [review]
